File: example/ios/PaymentsBridge.m
//
//  PaymentsBridge.m
//  MobilePaymentsSdkReactNativeExample
//
//  Created by Nilay Gupta on 07/10/24.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(PaymentsBridge, NSObject)
  RCT_EXTERN_METHOD(authorizePayments)
//  RCT_EXTERN_METHOD(classMethod2WithParams: (DataType) paramName)
@end

File: example/ios/PaymentsBridge.swift
import Foundation
import React
import SquareMobilePaymentsSDK

@objc(PaymentsBridge)
class PaymentsBridge: NSObject, RCTBridgeModule {  // Conform directly in the class declaration
    static func moduleName() -> String {
        return "PaymentsBridge"
    }

    @objc
  public func  authorizePayments() {
        let setupVC = SetupViewController()
        setupVC.authorizeMobilePaymentsSDK()
    }
}
